plugins/catalog-graphql/src/service/client.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import fetch from 'node-fetch';
+
+export class CatalogClient {
+  constructor(private baseUrl: string) {}
+  async list(): Promise<Entity[]> {
+    const res = await fetch(`${this.baseUrl}/catalog/entities`);
+    if (!res.ok) {
+      // todo(blam): need some better way to handle errors here
+      // experiment with throwing the input errors etc and having graphql versions of that
+      throw new Error(`NOPE, ${await res.text()}`);
```

Comment:
Maybe not necessary to worry too much about the handling here - if it's a non-200 it's pretty catastrophic so to speak. I think we could just improve the text of this error (include result code and a nice string like `Catalog request to http://localhost:7000/catalog/entities failed: 404 Not Found, <some payload text>`